pkg/arvo/sys/vane/gall.hoon

```diff
@@ -2,1457 +2,2882 @@
+    mo-state(moves [move moves])
   ::
-  ++  mo-okay                                           ::  valid agent core
-    ~/  %mo-okay
-    |=  vax/vase
+  ::  +mo-contains-valid-bowl: check that a vase contains a valid bowl.
+  ::
+  ++  mo-contains-valid-bowl
+    ~/  %mo-contains-valid-bowl
+    |=  =vase
     ^-  ?
-    =+  bol=(slew 12 vax)
-    ?~  bol  |
-    (~(nest ut p.u.bol) %| -:!>(*bowl))
-  ::  +mo-receive-core: receives an app core built by ford-turbo
+    ::
+    =/  inferred  -:!>(*bowl)
+    =/  maybe-vase  (slew 12 vase)
+    ::
+    ?~  maybe-vase
+      %.n
+    =/  =type  p.u.maybe-vase
+    (~(nest ut type) %.n inferred)
+  ::
+  ::  +mo-receive-core: receives an app core built by %ford.
+  ::
+  ::  Presuming we receive a good core, we first check to see if the agent
+  ::  is already running.  If so, we update its beak in %gall's state,
+  ::  initialise an +ap core for the agent, install the core we got from
+  ::  %ford, and then resolve any moves associated with it.
+  ::
+  ::  If we're dealing with a new agent, we create one using the result we got
+  ::  from %ford, add it to the collection of agents %gall is keeping track of,
+  ::  and then do more or less the same procedure as we did for the running
+  ::  agent case.
   ::
   ++  mo-receive-core
     ~/  %mo-receive-core
-    |=  [dap=dude byk=beak made-result=made-result:ford]
-    ^+  +>
+    |=  [=term =beak =made-result:ford]
+    ^+  mo-state
+    ::
+    ?:  ?=([%incomplete *] made-result)
+      (mo-give %onto %.n tang.made-result)
+    ::
+    =/  build-result  build-result.made-result
+    ::
+    ?:  ?=([%error *] build-result)
+      (mo-give %onto %.n message.build-result)
+    ::
+    =/  =cage  (result-to-cage:ford build-result)
+    =/  result-vase  q.cage
+    ::
+    =/  maybe-agent=(unit agent)
+      (~(get by running.agents.gall) term)
+    ::
+    ?^  maybe-agent
+      =/  agent  u.maybe-agent(beak beak)
+      ::
+      =.  running.agents.gall
+        (~(put by running.agents.gall) term agent)
+      ::
+      =/  =privilege
+        =/  =routes  [disclosing=~ attributing=our]
+        [%high routes]
+      ::
+      =/  app  (ap-abed:ap term privilege)
+      =.  app  (ap-reinstall:app result-vase)
+      ap-abet:app
+    ::
+    ?.  (mo-contains-valid-bowl result-vase)
+      =/  err  [[%leaf "{<term>}: bogus core"] ~]
+      (mo-give %onto %.n err)
+    ::
+    =.  mo-state  (mo-new-agent term beak result-vase)
+    ::
+    =/  old  mo-state
+    ::
+    =/  wag
+      =/  =routes  [disclosing=~ attributing=our]
+      =/  =privilege  [%high routes]
+      =/  app  (ap-abed:ap term privilege)
+      (ap-prep:app ~)
+    ::
+    =/  maybe-tang  -.wag
+    =/  app  +.wag
+    ::
+    ?^  maybe-tang
+      =.  mo-state  old
+      (mo-give %onto %.n u.maybe-tang)
+    ::
+    =.  mo-state  ap-abet:app
+    =.  mo-state  (mo-clear-queue term)
+    ::
+    =/  =suss  [term %boot now]
+    (mo-give %onto [%.y suss])
+  ::
```

Comment:
We don't usually include a vertical space above the first line of the comment, although it's not strictly disallowed.  The multiline body should be indented two more spaces, though.